sys/vm/vm_mmap.c

```diff
@@ -788,9 +792,9 @@ kern_mmap(struct thread *td, const struct mmap_req *mrp)
 			error = EINVAL;
 			goto done;
 		}
-		if ((max_prot & cap_maxprot) != max_prot) {
+		if ((cap_prot & cap_maxprot) != cap_prot) {
```

Comment:
I do wonder if we want to do a max_prot &= cap_maxprot after this point...

Reply:
it is only used as "max_prot & cap_maxprot" after this point

Reply:
I think @jrtc27's suggestion would be a good change.

Reply:
I implemented @jrtc27's suggestion on "max_prot & cap_maxprot"